addons/portal/static/src/chatter/frontend/portal_chatter_service.js

```diff
@@ -0,0 +1,103 @@
+import { PortalChatter } from "@portal/chatter/frontend/portal_chatter";
+import { App } from "@odoo/owl";
+import { getBundle } from "@web/core/assets";
+import { registry } from "@web/core/registry";
+import { rpc } from "@web/core/network/rpc";
+import { session } from "@web/session";
+import { _t } from "@web/core/l10n/translation";
+import { getTemplate } from "@web/core/templates";
+
+export class PortalChatterService {
+    constructor(env, services) {
+        this.setup(env, services);
+    }
+
+    setup(env, services) {
+        this.portalSecurity = {};
+        this.store = services["mail.store"];
+        this.busService = services.bus_service;
+    }
+
+    async createShadow(root) {
+        const shadow = root.attachShadow({ mode: "open" });
+        const res = await getBundle("portal.assets_chatter_style");
+        for (const url of res.cssLibs) {
+            const link = document.createElement("link");
+            link.rel = "stylesheet";
+            link.href = url;
+            shadow.appendChild(link);
+            await new Promise((res, rej) => {
+                link.addEventListener("load", res);
+                link.addEventListener("error", rej);
+            });
+        }
+        return shadow;
+    }
+
+    async initialize(env) {
+        const chatterEl = document.querySelector(".o_portal_chatter");
+        const props = {
+            resId: parseInt(chatterEl.getAttribute("data-res_id")),
+            resModel: chatterEl.getAttribute("data-res_model"),
+            portalSecurity: {
+                token: chatterEl.getAttribute("data-token"),
+                hash: chatterEl.getAttribute("data-hash"),
+                pid: parseInt(chatterEl.getAttribute("data-pid")),
+            },
```

Comment:
We usually avoid nested values (in particular in models, but it would make sense to not do it here either).